plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/RetryingConnectionFactory.java

```diff
@@ -18,33 +18,25 @@
 import dev.failsafe.Failsafe;
 import dev.failsafe.FailsafeException;
 import dev.failsafe.RetryPolicy;
-import io.trino.spi.TrinoException;
 import io.trino.spi.connector.ConnectorSession;
 
 import java.sql.Connection;
 import java.sql.SQLException;
-import java.sql.SQLRecoverableException;
+import java.sql.SQLTransientException;
 
-import static java.time.temporal.ChronoUnit.MILLIS;
-import static java.time.temporal.ChronoUnit.SECONDS;
 import static java.util.Objects.requireNonNull;
 
 public class RetryingConnectionFactory
         implements ConnectionFactory
 {
-    private static final RetryPolicy<Object> RETRY_POLICY = RetryPolicy.builder()
-            .withMaxDuration(java.time.Duration.of(30, SECONDS))
-            .withMaxAttempts(5)
-            .withBackoff(50, 5_000, MILLIS, 4)
-            .handleIf(RetryingConnectionFactory::isSqlRecoverableException)
-            .abortOn(TrinoException.class)
-            .build();
+    private final RetryPolicy<Object> retryPolicy;
 
     private final ConnectionFactory delegate;
 
     @Inject
-    public RetryingConnectionFactory(@ForRetryingConnectionFactory ConnectionFactory delegate)
+    public RetryingConnectionFactory(@ForRetryingConnectionFactory ConnectionFactory delegate, RetryPolicy<Object> retryPolicy)
```

Comment:
I am kind of not feeling that using failsafe library as part of the class API is good idea. How about depend only on `RetryStrategy` and leave to `RetryingConnectionFactory` decision what is used for retrying. It can use failsafe or something else

Reply:
Moved failsafe class out of the DI back to the internal details of the class